File: Day180_2025/Q180_Leetcode.java
import java.util.*;

class Q180_Leetcode {
    public int numSubseq(int[] nums, int target) {
        int mod = 1_000_000_007;
        Arrays.sort(nums);
        int n = nums.length;
        int[] pow2 = new int[n + 1];
        pow2[0] = 1;

        for (int i = 1; i <= n; i++) {
            pow2[i] = (pow2[i - 1] * 2) % mod;
        }

        int res = 0, l = 0, r = n - 1;
        while (l <= r) {
            if (nums[l] + nums[r] > target) {
                r--;
            } else {
                res = (res + pow2[r - l]) % mod;
                l++;
            }
        }

        return res;
    }
}

File: Day180_2025/Q180_Leetcode.py
from typing import List

class Solution:
    def numSubseq(self, nums: List[int], target: int) -> int:
        mod = 10**9 + 7
        nums.sort()
        n = len(nums)
        pow2 = [1] * (n + 1)
        for i in range(1, n + 1):
            pow2[i] = pow2[i - 1] * 2 % mod
        
        res = 0
        l, r = 0, n - 1
        while l <= r:
            if nums[l] + nums[r] > target:
                r -= 1
            else:
                res = (res + pow2[r - l]) % mod
                l += 1
        return res
